packages/core/admin/ee/server/services/review-workflows/validation.js

```diff
@@ -32,6 +33,11 @@ module.exports = ({ strapi }) => {
       if (stages.length > this.limits.stagesPerWorkflow) {
         throw new ValidationError(ERRORS.STAGES_LIMIT);
       }
+      // Validate stage names are not duplicated
+      const stageNames = stages.map((stage) => stage.name);
+      if (uniq(stageNames).length !== stageNames.length) {
+        throw new ValidationError(ERRORS.DUPLICATED_STAGE_NAME);
+      }
```

Comment:
Interesting we all have taken different approaches :D Jamie went for `findIndex`, I went for `.filter().length > 1` and you did lodash :D 

Looks good to me. Do you think this needs a test?

Reply:
ofc, I updated the tests !

Reply:
and my first approach was using a Set 🤣

Reply:
Which is also a very good idea!